core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -56,7 +58,10 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
    */
   def writeFully(channel: WritableByteChannel): Unit = {
     for (bytes <- getChunks()) {
-      while (bytes.remaining > 0) {
+      val capacity = bytes.limit()
+      while (bytes.position() < capacity && bytes.remaining() > 0) {
```

Comment:
why we need `bytes.position() < capacity`? isn't `bytes.remaining() > 0` enough?